alicloud/resource_alicloud_slb_load_balancer.go

```diff
@@ -0,0 +1,715 @@
+	"github.com/aliyun/terraform-provider-alicloud/alicloud/connectivity"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+)
+
+func resourceAlicloudSlbLoadBalancer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAlicloudSlbLoadBalancerCreate,
+		Read:   resourceAlicloudSlbLoadBalancerRead,
+		Update: resourceAlicloudSlbLoadBalancerUpdate,
+		Delete: resourceAlicloudSlbLoadBalancerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(5 * time.Minute),
+			Delete: schema.DefaultTimeout(9 * time.Minute),
+		},
+		Schema: map[string]*schema.Schema{
+			"internet": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+				Removed:  "Field 'internet' has been removed from provider version 1.122. Use 'address_type' replaces it.",
+			},
+			"address": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.Get("address_type").(string) == "internet" {
+						return true
+					}
+					return false
+				},
+			},
+			"address_ip_version": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ipv4", "ipv6"}, false),
+				Default:      "ipv4",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.Get("address_type").(string) == "intranet" {
+						return true
+					}
+					return false
+				},
+			},
+			"address_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Computed:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringInSlice([]string{"internet", "intranet"}, false),
+			},
+			"bandwidth": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      1,
+				ValidateFunc: validation.IntBetween(1, 1000),
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.Get("internet_charge_type").(string) == "PayByTraffic" {
+						return true
+					}
+					return false
+				},
+			},
+			"delete_protection": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"off", "on"}, false),
+				Default:      "off",
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					if d.Get("payment_type").(string) == "Subscription" {
+						return true
+					}
+					return false
+				},
+			},
+			"internet_charge_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"PayByBandwidth", "PayByTraffic"}, false),
```

Comment:
Please using `validation.StringInSlice([]string{"PayByBandwidth", "PayByTraffic"}, true)` attempt to correct the previous values. It aims to keep sync the value with OpenAPI parameter values.